tests/libvmi/vmi.go

```diff
@@ -88,9 +88,20 @@ func WithRng() Option {
 // WithResourceMemory specifies the vmi memory resource.
 func WithResourceMemory(value string) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
-			k8sv1.ResourceMemory: resource.MustParse(value),
+		if vmi.Spec.Domain.Resources.Requests == nil {
```

Comment:
conditionals make me worry. Can you add a longer explanation in the commit message why we need this logic? Do you want to reuse `WithResourceMemory` multiple times on the same built vmi? Is it a good idea?

Reply:
The reason for this change is if a vmi has a `k8sv1.ResourceCpu` set this function will delete it and i think that this is a bug.
I will handle this problem in another PR to make things simple.